engine/apps/user_management/tests/test_sync.py

```diff
@@ -20,6 +21,64 @@
 MOCK_GRAFANA_INCIDENT_BACKEND_URL = "https://grafana-incident.test"
 
 
+@contextmanager
+def patched_grafana_api_client(organization, is_rbac_enabled_for_organization=False):
+    GRAFANA_INCIDENT_PLUGIN_BACKEND_URL_KEY = "backendUrl"
+
+    with patch("apps.user_management.sync.GrafanaAPIClient") as mock_grafana_api_client:
+        mock_grafana_api_client.GRAFANA_INCIDENT_PLUGIN_BACKEND_URL_KEY = GRAFANA_INCIDENT_PLUGIN_BACKEND_URL_KEY
+
+        mock_client_instance = mock_grafana_api_client.return_value
+
+        mock_client_instance.get_users.return_value = [
+            {
+                "userId": 1,
+                "email": "test@test.test",
+                "name": "Test",
+                "login": "test",
+                "role": "admin",
+                "avatarUrl": "test.test/test",
+                "permissions": [],
+            },
+        ]
+        mock_client_instance.get_teams.return_value = (
+            {
+                "totalCount": 1,
+                "teams": (
+                    {
+                        "id": 1,
+                        "name": "Test",
+                        "email": "test@test.test",
+                        "avatarUrl": "test.test/test",
+                    },
+                ),
+            },
+            None,
+        )
+        mock_client_instance.get_team_members.return_value = (
+            [
+                {
+                    "orgId": organization.org_id,
+                    "teamId": 1,
+                    "userId": 1,
+                },
+            ],
+            None,
+        )
+        mock_client_instance.get_grafana_incident_plugin_settings.return_value = (
+            {"enabled": True, "jsonData": {GRAFANA_INCIDENT_PLUGIN_BACKEND_URL_KEY: MOCK_GRAFANA_INCIDENT_BACKEND_URL}},
+            None,
+        )
+        mock_client_instance.get_grafana_labels_plugin_settings.return_value = (
+            {"enabled": True, "jsonData": {}},
+            None,
+        )
+        mock_client_instance.check_token.return_value = (None, {"connected": True})
+        mock_client_instance.is_rbac_enabled_for_organization.return_value = is_rbac_enabled_for_organization
+
+        yield mock_grafana_api_client
```

Comment:
most of the changes in this file are related to this..

we had a lot of duplication around mocking out responses on the `GrafanaAPIClient` (plus several instances where we had like 8 nested context managers 😆 ).. this gets rid of that